library.c

```diff
@@ -1798,65 +2104,45 @@ redis_sock_create(char *host, int host_len, int port,
     redis_sock->compression = REDIS_COMPRESSION_NONE;
     redis_sock->compression_level = 0; /* default */
     redis_sock->mode = ATOMIC;
-    redis_sock->head = NULL;
-    redis_sock->current = NULL;
-
-    redis_sock->pipeline_cmd = NULL;
-
-    redis_sock->err = NULL;
 
     redis_sock->scan = 0;
-
```

Comment:
You removed all `NULL` initializers, maybe all `0` are redundant and should be deleted too?